Ladybird/MachPortServer.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Platform.h>
+
+#if !defined(AK_OS_MACH)
+#    error "This file is only for Mach kernel-based OS's"
+#endif
+
+#include <AK/Atomic.h>
+#include <AK/String.h>
+#include <LibCore/MachPort.h>
+#include <LibThreading/Thread.h>
+
+namespace Ladybird {
+
+class MachPortServer {
+
+public:
+    MachPortServer();
+    ~MachPortServer();
+
+    void start();
+    void stop();
+
+    bool is_initialized();
+
+    Function<void(pid_t, Core::MachPort)> on_receive_child_mach_port;
+
+    ByteString const& server_port_name() const { return m_server_port_name; }
+
+private:
+    void thread_loop();
+    ErrorOr<void> allocate_server_port();
+
+    NonnullRefPtr<Threading::Thread> m_thread;
+    ByteString const m_server_port_name;
```

Comment:
Not `String`?

Reply:
It needs to be null terminated to pass to the boostrap_xxx apis

Reply:
I thought it was something like that, but `register_with_bootstrap_server` takes a StringView and copies to a `name_t bootstrap_name` buffer

Reply:
Eh, true. we probably don't actually need a name_t buffer. all the other code around the bootstrap name assumes ByteString so I'll just make it consistent.